chain-extensions/dapps-staking/src/lib.rs

```diff
@@ -65,19 +64,27 @@ impl TryFrom<u32> for DappsStakingFunc {
     }
 }
 
-pub struct DappsStakingExtension<R>(PhantomData<R>);
+/// Dapps Staking chain extension.
+pub struct DappsStakingExtension<T>(PhantomData<T>);
 
-impl<T: pallet_dapps_staking::Config> ChainExtensionExec<T> for DappsStakingExtension<T> {
-    fn execute_func<E>(
-        func_id: u32,
-        env: Environment<E, InitState>,
-    ) -> Result<RetVal, DispatchError>
+impl<T> Default for DappsStakingExtension<T> {
+    fn default() -> Self {
+        DappsStakingExtension(PhantomData)
+    }
+}
+
+impl<T> ChainExtension<T> for DappsStakingExtension<T>
+where
+    T: pallet_dapps_staking::Config + pallet_contracts::Config,
+    <T as pallet_dapps_staking::Config>::SmartContract: From<[u8; 32]>,
```

Comment:
I agree this looks more elegant but I'm not fond of assuming that this will be the only `256` bit value (even though it probably will be). Seems just a little bit hacky.

Reply:
Technically we can try to release it to `SysConfig::AccountId` type, let me try